homeassistant/components/aurora_abb_powerone/sensor.py

```diff
@@ -5,52 +5,99 @@
 from aurorapy.client import AuroraError, AuroraSerialClient
 import voluptuous as vol
 
-from homeassistant.components.sensor import PLATFORM_SCHEMA, SensorEntity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
 from homeassistant.const import (
     CONF_ADDRESS,
     CONF_DEVICE,
     CONF_NAME,
+    CONF_PORT,
     DEVICE_CLASS_POWER,
+    DEVICE_CLASS_TEMPERATURE,
     POWER_WATT,
+    TEMP_CELSIUS,
 )
+from homeassistant.exceptions import InvalidStateError
 import homeassistant.helpers.config_validation as cv
 
+from .aurora_device import AuroraDevice
+from .const import DEFAULT_ADDRESS, DOMAIN
+
 _LOGGER = logging.getLogger(__name__)
 
-DEFAULT_ADDRESS = 2
-DEFAULT_NAME = "Solar PV"
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_DEVICE): cv.string,
         vol.Optional(CONF_ADDRESS, default=DEFAULT_ADDRESS): cv.positive_int,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_NAME, default="Solar PV"): cv.string,
     }
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Aurora ABB PowerOne device."""
+def setup_platform(hass, config: dict, add_entities, discovery_info=None):
+    """Set up single sensor based on configuration.yaml (DEPRECATED)."""
     devices = []
     comport = config[CONF_DEVICE]
-    address = config[CONF_ADDRESS]
-    name = config[CONF_NAME]
-
+    address = config.get(CONF_ADDRESS, DEFAULT_ADDRESS)
+    name = config.get(CONF_NAME, "Solar PV")
+    _LOGGER.warning(
+        "DEPRECATED: setting up %s via configuration.yaml will "
+        "soon be unsupported.  Please remove the entries from your "
+        "configuration.yaml file and then set up the integration via the UI",
+        DOMAIN,
+    )
     _LOGGER.debug("Intitialising com port=%s address=%s", comport, address)
```

Comment:
Import the data into a config entry with SOURCE_IMPORT instead

Reply:
I agree that this can be done.  But there are are a very small number of users using this integration.  Is it worth the extra code for this?

Reply:
Now implemented.